src/main/java/com/patreon/PatreonAPI.java

```diff
@@ -0,0 +1,151 @@
+     */
+    public JSONAPIDocument<List<Campaign>> fetchCampaigns() throws IOException {
+        String path = new URIBuilder()
+            .setPath("current_user/campaigns")
+            .addParameter("include", "rewards,creator,goals")
+            .toString();
+        return converter.readDocumentCollection(
+            getDataStream(path),
+            Campaign.class
+        );
+    }
+
+    /**
+     * Retrieve pledges for the specified campaign
+     *
+     * @param campaignId id for campaign to retrieve
+     * @param pageSize   how many pledges to return
+     * @param pageCursor ignore, put null.
+     * @return JSONAPIDocument<List<Pledge>> containing pledges & associated data
+     * @throws IOException Thrown when the GET request failed
+     */
+    public JSONAPIDocument<List<Pledge>> fetchPageOfPledges(String campaignId, int pageSize, String pageCursor) throws IOException {
+        URIBuilder pathBuilder = new URIBuilder()
+            .setPath(String.format("campaigns/%s/pledges", campaignId))
+            .addParameter("page[count]", String.valueOf(pageSize));
+        if (pageCursor != null) {
+            pathBuilder.addParameter("page[cursor]", pageCursor);
+        }
+        return converter.readDocumentCollection(
+            getDataStream(pathBuilder.toString()),
+            Pledge.class
+        );
+    }
+
+    public String getCursorFromPledgesResponse(JSONAPIDocument<List<Pledge>> pledgesResponse) {
+        Links pledgesLinks = pledgesResponse.getLinks();
+        if (pledgesLinks == null) {
+            return null;
+        }
+        Link nextLink = pledgesLinks.getNext();
+        if (nextLink == null) {
+            return null;
+        }
+        String nextLinkString = nextLink.toString();
+        try {
+            List<NameValuePair> queryParameters = URLEncodedUtils.parse(new URI(nextLinkString), "utf8");
+            for (NameValuePair pair : queryParameters) {
+                String name = pair.getName();
+                if (name.equals("page[cursor]")) {
+                    String cursorValue = pair.getValue();
+                    return cursorValue;
+                }
+            }
+        } catch (URISyntaxException e) {
+            LOG.error(e.getMessage());
+        }
+        return null;
+    }
+
+    public List<Pledge> fetchAllPledges(String campaignId) throws IOException {
+        Set<Pledge> pledges = new HashSet<>();
+        String cursor = null;
+        while (true) {
+            JSONAPIDocument<List<Pledge>> pledgesPage = fetchPageOfPledges(campaignId, 5, cursor);
+            pledges.addAll(pledgesPage.get());
+            cursor = getCursorFromPledgesResponse(pledgesPage);
+            if (cursor == null) {
+                break;
+            }
+        }
+        return new ArrayList<>(pledges);
+    }
+
+    private InputStream getDataStream(String suffix) throws IOException {
+        try {
+            String prefix = "https://api.patreon.com/oauth2/api/";
```

Comment:
think this should be `www.patreon.com/api`

Reply:
i'd really love to keep our platform API at `api.patreon.com` and have `www.patreon.com/api` be an in-house implementation detail for our js clients